### PR TITLE
Add DataToolParameter back to parameters

### PR DESCRIPTION
**Pull request by guerler**

### CODE REVIEW

**Comment by martenson** [comment]
xref https://github.com/galaxyproject/galaxy/issues/2152

File: lib/galaxy/tools/parameters/__init__.py [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
 
-from basic import RuntimeValue
+from basic import RuntimeValue, DataToolParameter
 from grouping import Conditional, Repeat, Section, UploadDataset
 from galaxy.util.expressions import ExpressionContext
 from galaxy.util.json import json_fix
 
 REPLACE_ON_TRUTHY = object()
 
+def export():
```
Comment (dannon) [inline lib/galaxy/tools/parameters/__init__.py]: This bit seems weird to me, how is this used downstream?

(see toplevel pr comment, got info about why it was added on irc)

**Comment by dannon** [comment]
(chatted with @guerler briefly offline to understand the export function).  How about we write an `__all__` module definition here instead?

**Comment by nsoranzo** [comment]
This should target 16.04 (or backported later).

**Comment by guerler** [comment]
Thanks @dannon.

**Comment by guerler** [comment]
Close in favor of #2156

**Pull request closed without merging**